### ISSUE TITLE
capstone.capstone.CsError Different API version between core & binding

### ISSUE THREAD

**Issue by nap82** [secretsquirrel/the-backdoor-factory#40]
I am using Kali Linux v1.1.0 kernel 3.18.0 and used apt-get install backdoor-factory to install the tool.  Whenever I run backdoor-factory it comes back with "capstone.capstone.CsError Different API version between core & binding (CS_ERR_VERSION).  I've attached a screenshot of what I'm seeing with specifics.  Also, I've run the tool with the -S switch (psexec.exe, wget.exe, and plink.exe) and none comes back saying it's supported.  Is there a listing of supported [.exe] files, or is this most likely related to the error I'm seeing as well?

![screen shot 2015-03-18 at 3 10 54 pm](https://cloud.githubusercontent.com/assets/4315051/6718349/1a21e158-cd81-11e4-85e5-bafcf0153adb.png)

**Comment by secretsquirrel** [comment]
All those binaries are supported.  It seems you have multiple versions of capstone installed on your machine.

Do the following and paste the results:

$python
>>>import capstone
>>>print capstone.version_bind()

nap82 wrote:

**Comment by nap82** [comment]
Thanks for writing back.  I ran the commands and here's the output:  (4, 0, 1024)
 I was able to do a bit of troubleshooting and the problem goes back to an unsupported configuration - I installed veil-evasion on Kali after I installed backdoor-factory.  I replicated it in a vm with the same results.  If there's a quick fix I can do to repair having two versions of capstone I'd much appreciate it.  I'll continue to do some more troubleshooting.

**Comment by secretsquirrel** [comment]
BDF works only with capstone 3, for is the next version I suppose.


On Mar 18, 2015, at 8:02 PM, nap82 <notifications@github.com> wrote:

Thanks for writing back. I ran the commands and here's the output: (4, 0,
1024)
I was able to do a bit of troubleshooting and the problem goes back to an
unsupported configuration - I installed veil-evasion on Kali after I
installed backdoor-factory. I replicated it in a vm with the same results.
If there's a quick fix I can do to repair having two versions of capstone
I'd much appreciate it. I'll continue to do some more troubleshooting.

—
Reply to this email directly or view it on GitHub
<https://github.com/secretsquirrel/the-backdoor-factory/issues/40#issuecomment-83228690>
.

**Comment by nap82** [comment]
The solution I came to, and was able to get to work fine, is to uninstall both BDF and veil-evasion completely.  Then I re-installed veil-evasion.  I will just run BDF (at this time) from within veil-evasion.  Here's what I had to do to uninstall both packages and dependencies.

rm -rf /usr/share/backdoor-factory
rm -rf /usr/share/veil-evasion
apt-get -y remove backdoor-factory
apt-get -y remove veil-evasion
apt-get -y remove python-capstone
